### PR TITLE
Make requests to web-server in supervisor

### PR DESCRIPTION
**Pull request by bklvsky**
- Request tasks at `build_node/get_task` endpoint from supervisor
- Remove requesting new task from builders

Resolves: AlmaLinux/build-system#371

### CODE REVIEW

File: almalinux_build_node.py [hunk 1]
```diff
@@ -157,13 +169,25 @@ def sigusr_handler(signum, frame):
 
     node_globals.init_supervisors(config)
     builders = []
+    task_queue = queue.Queue()
```
Comment (Korulag) [inline almalinux_build_node.py:172]: IMO, you need to limit queue size to the amount of workers + 1 or just some fixed size that is configurable and do not request for new tasks when it's full

File: build_node/build_node_supervisor.py [hunk 2]
```diff
@@ -47,18 +98,12 @@ def run(self):
...
-                self.__session.post(
-                    full_url, json=data, timeout=self.config.request_timeout)
-            except Exception:
-                logging.error(
-                    "Can't report active task to master:\n%s",
-                    traceback.format_exc()
-                )
-            self.terminated_event.wait(60)
+            self.__report_active_tasks()
+            task = self.__request_build_task()
```
Comment (Korulag) [inline build_node/build_node_supervisor.py:102]: If implemented the thing from above, please put check for queue being full here and only call for new task when there are empty slots

**Review by Korulag — CHANGES_REQUESTED** [review]

File: almalinux_build_node.py (hunk 1 above)
Comment (bklvsky) [reply almalinux_build_node.py]: done

File: almalinux_build_node.py [hunk 3]
```diff
@@ -93,12 +97,16 @@ def init_working_dir(config):
                 #   rm: cannot remove 'test/test.txt': Permission denied
                 rm_sudo(os.path.join(working_dir, name))
     else:
-        logging.debug('creating the {0} working directory'.
-                      format(config.working_dir))
+        logging.debug(
+            'creating the {0} working directory'.format(config.working_dir)
```
Comment (javihernandez) [inline almalinux_build_node.py:101]: Not critical, but I'd prefer to use either %-syntax or f-strings when logging stuff - same applies to others around

File: almalinux_build_node.py [hunk 4]
```diff
@@ -157,13 +169,25 @@ def sigusr_handler(signum, frame):
...
     node_globals.init_supervisors(config)
     builders = []
+    task_queue = queue.Queue(maxsize=config.queue_size)
+
     for i in range(0, config.threads_count):
-        builder = BuildNodeBuilder(config, i, node_terminated,
-                                   node_graceful_terminated)
+        builder = BuildNodeBuilder(
+            config,
+            i,
```
Comment (javihernandez) [inline almalinux_build_node.py:177]: Using single char variable names when iterating is fine, however, in this particular use case and for readability purposes, I'd call it `thread_num` or something like that.

File: almalinux_build_node.py (hunk 3 above)
Comment (bklvsky) [reply almalinux_build_node.py:101]: yeah, i thought the same but it's everywhere in the old code. maybe i should create a separate task to get rid of this kind of formatting all over the albs?

File: build_node/build_node_builder.py [hunk 5]
```diff
@@ -57,11 +62,13 @@ def __init__(self, config, thread_num, terminated_event,
         graceful_terminated_event : threading.Event
             Shows, if process got "kill -10" signal.
         """
-        super(BuildNodeBuilder, self).__init__(name='Builder-{0}'.format(
-            thread_num))
+        super(BuildNodeBuilder, self).__init__(
```
Comment (javihernandez) [inline build_node/build_node_builder.py:65]: In modern python we can do `super().method(arg)`, see https://docs.python.org/3/library/functions.html#super

Comment (bklvsky) [reply build_node/build_node_builder.py:65]: same as the above for string formatting, currently it's everywhere in older code. we could do one task for all reformatting to update the codebase

**Review by javihernandez — APPROVED** [review]

**Review by anfimovdm — APPROVED** [review]

**Review by maccelf — APPROVED** [review]

**Pull request merged**